pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -276,7 +276,7 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 			err = og.verifyVolumeIsSafeToDetach(volumeToDetach)
 		}
 		if err == nil {
-			err = volumeDetacher.Detach(volumeName, volumeToDetach.NodeName)
+			err = volumeDetacher.Detach(volumeToDetach.VolumeSpec, volumeName, volumeToDetach.NodeName)
```

Comment:
You got the reason right!

Unless you have a better idea (?), I believe that this imperfection can be tolerated for now given the raw improvement that this PR provides. If this becomes an actual issue for any user, then we could brainstorm harder and follow up with a smarter solution in due time. What do you think?